### PR TITLE
Add a pool of evhttp_connection.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: cpp/net/connection_pool.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  evhttp_connection_unique_ptr Get(const URL& url);
+  void Put(evhttp_connection_unique_ptr&& conn);
+
+ private:
+  libevent::Base* const base_;
+
+  std::mutex lock_;
+  // We get and put connections from the back of the deque, and when
+  // there are too many, we prune them from the front (LIFO).
+  std::map<std::pair<std::string, uint16_t>,
```
Comment (AlCutter) [inline cpp/net/connection_pool.h]: Could typedef `std::pair<std::string, uint16_t>` to a `HostPortPair` or similar for a small increase in readability and longer life for our `:`, `<` and `>` keys.

File: cpp/net/connection_pool.cc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+using std::mutex;
+using std::pair;
+using std::string;
+
+namespace cert_trans {
+namespace internal {
+
+namespace {
+
+const int kMaxConnPerHostPort = 4;
```
Comment (AlCutter) [inline cpp/net/connection_pool.cc]: flag?

File: cpp/net/connection_pool.cc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  it->second.pop_back();
+
+  return retval;
+}
+
+
+void ConnectionPool::Put(evhttp_connection_unique_ptr&& conn) {
+  char* host;
+  uint16_t port;
+  evhttp_connection_get_peer(conn.get(), &host, &port);
```
Comment (AlCutter) [inline cpp/net/connection_pool.cc]: `CHECK(conn)` ?
Shouldn't really happen, but...

Comment (pphaneuf) [inline cpp/net/connection_pool.cc]: I think silently ignoring that is the best course of action.

**Comment by pphaneuf** [comment]
Comments addressed.

**Comment by AlCutter** [comment]
LGTM

**Pull request merged**